primefaces/src/main/java/org/primefaces/component/api/UIData.java

```diff
@@ -548,19 +548,9 @@ protected void saveInputState(EditableValueHolder input, FacesContext context) {
             return;
         }
 
-        SavedState state = null;
-
-        if (saved == null) {
-            state = new SavedState();
-        }
-
-        if (state == null) {
-            state = saved.get(componentClientId);
-
-            if (state == null) {
-                state = new SavedState();
-            }
-        }
+        SavedState state = saved != null
+                ? saved.computeIfAbsent(componentClientId, key -> new SavedState())
```

Comment:
is the "put" by design? your old code did not saved.put(..)

Reply:
It is put below (line 559 currently)